src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,48 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final int DATE_OF_PAY = 10;
+    private static final int HOURS_WORKED = 2;
+    private static final int SALARY_PER_HOUR = 3;
+    private static final int START_OF_RANGE = 0;
+    private static final String SPACE = " ";
+    private static final String REPORT_MESSAGE = "Report for period ";
+    private static final String SEPARATE = " - ";
+    private StringBuilder sb = new StringBuilder();
+    private int salary = 0;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate dateStart = LocalDate.parse(dateFrom, FORMATTER);
+        LocalDate dateLast = LocalDate.parse(dateTo, FORMATTER);
+        String[] resultToReport = new String[names.length];
+
+        for (int i = 0; i < names.length; i++) {
+            salary = 0;
+            for (String dataArr : data) {
+                if (dataArr.contains(names[i])) {
```

Comment:
Can we choose another name for variable `dataArr`?
because it is a `String` variable, not an array